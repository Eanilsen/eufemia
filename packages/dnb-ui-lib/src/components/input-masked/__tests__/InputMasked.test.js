/**
 * Component Test
 *
 */

import React from 'react'
import {
  shallow,
  fakeProps,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../InputMasked'

// just to make sure we re-run the test in watch mode due to changes in theese files
import _input_masked from '../style/_input-masked.scss' // eslint-disable-line
import dnb_input_masked from '../style/dnb-input-masked.scss' // eslint-disable-line

const props = {
  ...fakeProps(require.resolve('../InputMasked'), {
    optional: true
  }),
  disabled: false
}

describe('InputMasked component', () => {
  // shallow compare the snapshot
  it('have to match type="text" snapshot', () => {
    const Comp = shallow(<Component {...props} type="text" value="test" />)
    expect(toJson(Comp)).toMatchSnapshot()
  })
})

describe('InputMasked scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/dnb-input-masked.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
