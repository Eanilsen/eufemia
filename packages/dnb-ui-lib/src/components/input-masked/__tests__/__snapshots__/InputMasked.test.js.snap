// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMasked component have to match type="text" snapshot 1`] = `
<Input
  align={null}
  attributes={null}
  autocomplete="off"
  class={null}
  className={null}
  custom_element={null}
  custom_method={null}
  description={null}
  disabled={false}
  extra_information={null}
  font_class={null}
  id={null}
  inputElement={[Function]}
  mask={Array []}
  on_blur={null}
  on_change={null}
  on_focus={null}
  on_submit={null}
  on_submit_blur={null}
  on_submit_focus={null}
  placeholder={null}
  search_button_title=""
  show_mask="show_mask"
  size={null}
  type="text"
  value="test"
/>
`;

exports[`InputMasked scss have to match snapshot 1`] = `
"/*
* DNB Input
*
* ======================================================================== */
/*
* Colors
* ======================================================================== */
/*
 * Typography Family
 * ======================================================================== */
/*
* Typography Size
* ======================================================================== */
/*
* Grid and units
* ======================================================================== */
/*
* Breakpoints
* ======================================================================== */
/*
 * Utilities
 * ========================================================================== */
.dnb-sr-only {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px; }

.dnb-not-sr-only {
  position: static; }

.dnb-unstyled-list {
  margin: 0;
  list-style-type: none;
  padding-left: 0; }
  .dnb-unstyled-list > li {
    margin-bottom: 0; }

.dnb-no-focus {
  outline: none; }

.dnb-fake-focus {
  outline: none; }
  .dnb-fake-focus:focus {
    outline: none;
    border-color: transparent; }
    html[data-whatinput='mouse'] .dnb-fake-focus:focus:not(.dnb-no-mouse-focus) {
      transition: box-shadow 0.1s ease-out 0s;
      box-shadow: 0 0 0 2px #14555a; }
    html[data-whatinput='keyboard'] .dnb-fake-focus:focus {
      box-shadow: 0 0 0 2px #ff5400; }

@media screen and (max-width: 700px) {
  .dnb-hide-on-mobile {
    display: none; } }

@media screen and (min-width: 700px) {
  .dnb-mobile-exclusive {
    display: none; } }

.dnb-width-limit {
  margin: 0 auto;
  width: 100%;
  max-width: 1024px;
  padding: 0 0.5rem; }
  @media screen and (min-width: 700px) {
    .dnb-width-limit {
      padding-left: 1rem;
      padding-right: 1rem; } }

.dnb-belt {
  padding: 1rem 0;
  width: 100%;
  background: #fafafa; }

.dnb-nudge {
  display: inline-flex;
  flex: 0 0 100%; }
  .dnb-nudge--vertical[data-nudges='1'] {
    height: 1rem; }
  .dnb-nudge--horizontal[data-nudges='1'] {
    width: 1rem; }
  .dnb-nudge--vertical[data-nudges='2'] {
    height: 2rem; }
  .dnb-nudge--horizontal[data-nudges='2'] {
    width: 2rem; }
  .dnb-nudge--vertical[data-nudges='3'] {
    height: 3rem; }
  .dnb-nudge--horizontal[data-nudges='3'] {
    width: 3rem; }
  .dnb-nudge--vertical[data-nudges='4'] {
    height: 4rem; }
  .dnb-nudge--horizontal[data-nudges='4'] {
    width: 4rem; }
  .dnb-nudge--vertical[data-nudges='5'] {
    height: 5rem; }
  .dnb-nudge--horizontal[data-nudges='5'] {
    width: 5rem; }
  .dnb-nudge--vertical[data-nudges='6'] {
    height: 6rem; }
  .dnb-nudge--horizontal[data-nudges='6'] {
    width: 6rem; }
  .dnb-nudge--vertical[data-nudges='7'] {
    height: 7rem; }
  .dnb-nudge--horizontal[data-nudges='7'] {
    width: 7rem; }
  .dnb-nudge--vertical[data-nudges='8'] {
    height: 8rem; }
  .dnb-nudge--horizontal[data-nudges='8'] {
    width: 8rem; }
  .dnb-nudge--vertical[data-nudges='9'] {
    height: 9rem; }
  .dnb-nudge--horizontal[data-nudges='9'] {
    width: 9rem; }
  .dnb-nudge--vertical[data-nudges='10'] {
    height: 10rem; }
  .dnb-nudge--horizontal[data-nudges='10'] {
    width: 10rem; }

/*
* Scopes
* ======================================================================== */
/*
* DNB Input
*
* ======================================================================== */
/*
* DNB font-import from components
* This file is totaly the same as \\"core/_font-import.scss\\"
* but has a difference back path
*
* ======================================================================== */
@font-face {
  font-family: 'Fedra Sans Std Book';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std Book';
  src: url(\\"../../../../assets/fonts/FedraSansStd-BookItalic.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-BookItalic.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-BookItalic.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-BookItalic.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: italic; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Light.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Light.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Light.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Light.ttf\\") format(\\"truetype\\");
  font-weight: 300;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-LightItalic.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-LightItalic.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-LightItalic.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-LightItalic.ttf\\") format(\\"truetype\\");
  font-weight: 300;
  font-style: italic; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Medium.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Medium.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Medium.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Medium.ttf\\") format(\\"truetype\\");
  font-weight: 500;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-MediumItalic.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-MediumItalic.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-MediumItalic.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-MediumItalic.ttf\\") format(\\"truetype\\");
  font-weight: 500;
  font-style: italic; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Bold.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Bold.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Bold.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Bold.ttf\\") format(\\"truetype\\");
  font-weight: bold;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts/FedraSansStd-BoldItalic.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-BoldItalic.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-BoldItalic.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-BoldItalic.ttf\\") format(\\"truetype\\");
  font-weight: bold;
  font-style: italic; }

/*
* Forms and inputs
* ======================================================================== */
legend {
  padding: 0;
  display: table;
  width: 100%; }

fieldset {
  border: 0;
  padding: 0.01em 0 0 0;
  margin: 0;
  min-width: 0; }

body:not(:-moz-handler-blocked) fieldset {
  display: table-cell; }

/*
* DNB icon
*
* ======================================================================== */
/*
* Icon component
*
* ======================================================================== */
.dnb-icon {
  display: inline-block;
  font-size: 0;
  width: 100%; }
  .dnb-icon svg {
    fill: currentColor; }

.dnb-input {
  font-size: 1rem;
  font-family: \\"Fedra Sans Std\\", sans-serif;
  font-weight: 300;
  font-style: normal;
  line-height: 1.5em; }
  .dnb-input * {
    margin: 0;
    padding: 0; }
  .dnb-input *,
  .dnb-input ::before,
  .dnb-input ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-input ::before,
  .dnb-input ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
* Input component
*
* ======================================================================== */
.dnb-input {
  display: flex;
  align-items: center;
  flex-flow: row wrap;
  width: 100%; }
  .dnb-input--label-hidden label {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px; }
  .dnb-input__shell {
    position: relative;
    flex: 1;
    display: flex;
    width: 100%;
    height: 100%;
    margin: 0;
    align-items: center;
    background-color: #f5fafa;
    border-radius: 2rem;
    border: 1px solid #bedbdb;
    transition: background-color 0.5s ease-out 0s; }
    .no-touch .dnb-input__shell :not([disabled]):hover[disabled] {
      cursor: not-allowed; }
    .no-touch .dnb-input__shell :not([disabled]):hover:not([disabled]) {
      background-color: rgba(255, 255, 255, 0.7); }
  .dnb-input__description, .dnb-input__placeholder, .dnb-input__input {
    font-size: 1rem;
    line-height: 1.25; }
  .dnb-input__placeholder, .dnb-input__input {
    width: 100%;
    height: 2.2rem;
    padding: 1.2rem 0 1rem 1rem;
    font-family: inherit;
    font-weight: inherit;
    outline: none;
    background-color: transparent;
    color: inherit;
    border: none;
    border-radius: 2rem; }
  .dnb-input__input[disabled] {
    opacity: 0.5;
    user-select: none; }
  .dnb-input__placeholder {
    pointer-events: none;
    position: absolute;
    left: 0;
    top: 2px;
    padding: 0 1rem;
    height: 100%;
    overflow: hidden;
    white-space: nowrap;
    opacity: 1;
    color: rgba(51, 51, 51, 0.8);
    line-height: 2.2rem;
    text-overflow: ellipsis;
    border-radius: 0;
    transition: opacity 0.1s linear 0s; }
  .dnb-input__description {
    margin: 0 0 0 0.5rem; }
  .dnb-input__extra-information {
    flex: 1 100%;
    margin: 0.5rem 0 0 1rem;
    font-size: 0.75rem;
    line-height: 1rem; }
  .dnb-input--large .dnb-input__description,
  .dnb-input--large .dnb-input__input,
  .dnb-input--large .dnb-input__placeholder {
    font-size: 2rem; }
  .dnb-input--large .dnb-input__placeholder {
    line-height: 3rem; }
  .dnb-input--large .dnb-input__input {
    height: 3rem;
    padding: 1.5rem 0 1.5rem 1.5rem;
    border-radius: 3rem; }
  .dnb-input__align--right {
    padding-left: 0 !important;
    padding-right: 1rem !important;
    text-align: right; }
  .dnb-input__input--search {
    border-radius: 2rem 0 0 2rem;
    margin: 0 1px 0 0; }
  .dnb-input__search-submit {
    position: relative;
    top: -1px;
    right: -1px;
    /* stylelint-disable */
    flex: 0 0 2.2rem;
    flex: 0 0 calc(4px + 2.2rem);
    height: 2.2rem;
    height: calc(4px + 2.2rem);
    /* stylelint-enable */
    margin: 0 0 -2px;
    overflow: hidden;
    border-radius: 2rem;
    border: 1px solid #bedbdb; }
  .dnb-input__search-submit__button {
    position: relative;
    cursor: pointer;
    display: flex;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 0;
    outline: none;
    border: none;
    background-color: white;
    color: #007272; }
  .dnb-input__shell[data-input-state='focus'] {
    border-color: #a0d6c9;
    background-color: white; }
    .dnb-input__shell[data-input-state='focus'] .dnb-input__search-submit {
      border-color: rgba(210, 239, 232, 0.5); }
  .dnb-input__shell[data-has-content='true'] .dnb-input__placeholder,
  .dnb-input__shell[data-input-state='focus'][data-has-content='true'] .dnb-input__placeholder {
    display: none; }
  .dnb-input__shell[data-input-state='focus'][data-has-content='false'] .dnb-input__placeholder {
    opacity: 0.35; }

.dnb-input__search-submit[data-input-state='focus'] {
  border-color: #bedbdb; }

.dnb-input__shell[data-has-content='true'] {
  background-color: rgba(255, 255, 255, 0.7); }

.dnb-input__shell[data-input-state='focus'],
.dnb-input__search-submit[data-input-state='focus'] {
  box-shadow: 0 0 0 2px #007272;
  outline: none;
  border-color: transparent; }
  html[data-whatinput='mouse'] .dnb-input__shell[data-input-state='focus']:not(.dnb-no-mouse-focus), html[data-whatinput='mouse']
  .dnb-input__search-submit[data-input-state='focus']:not(.dnb-no-mouse-focus) {
    transition: box-shadow 0.1s ease-out 0s;
    box-shadow: 0 0 0 2px #14555a; }
  html[data-whatinput='keyboard'] .dnb-input__shell[data-input-state='focus'], html[data-whatinput='keyboard']
  .dnb-input__search-submit[data-input-state='focus'] {
    box-shadow: 0 0 0 2px #ff5400; }

/*
* Range input
* ======================================================================== */
.dnb-range-slider__css-vars__bg-color-1 {
  background-color: rgba(0, 114, 114, 0.5); }

.dnb-range-slider__css-vars__bg-color-2 {
  background-color: #28b482; }

input[type='range'] {
  -webkit-appearance: none;
  width: 100%;
  border-radius: 0.25rem; }

input[type='range']::-webkit-slider-runnable-track {
  position: relative;
  z-index: 1;
  cursor: pointer;
  width: 100%;
  height: 0.25rem;
  box-shadow: none; }

input[type='range']::-webkit-slider-thumb {
  position: relative;
  top: 0.125rem;
  cursor: pointer;
  height: 1.5rem;
  width: 1.5rem;
  border-radius: 1.5rem;
  background: white;
  border: 1px solid #98c6c6;
  transform: translate(0, -50%);
  -webkit-appearance: none;
  box-shadow: 1px 1px 2px rgba(0, 135, 137, 0.25), 0 0 1px rgba(0, 160, 163, 0.25); }

input[type='range']:focus {
  outline: none; }

.dnb-input-masked {
  font-size: 1rem;
  font-family: \\"Fedra Sans Std\\", sans-serif;
  font-weight: 300;
  font-style: normal;
  line-height: 1.5em; }
  .dnb-input-masked * {
    margin: 0;
    padding: 0; }
  .dnb-input-masked *,
  .dnb-input-masked ::before,
  .dnb-input-masked ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-input-masked ::before,
  .dnb-input-masked ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
* InputMasked component
*
* ======================================================================== */
"
`;
